### PR TITLE
Feat 152 text input component

### PR DESCRIPTION
**Pull request by melaniebmn**
# Summary | Résumé

Adding new input component, including error message, hint, and label.

Ticket: https://app.zenhub.com/workspaces/design-system-6100624a19f4cf000e46e458/issues/cds-snc/design-gc-conception/159

### CODE REVIEW

**Review by ethanWallace — CHANGES_REQUESTED** [review]
It's looking good! Just a few accessibility things to fix up and it should be good to go.

**Comment by ethanWallace** [comment]
Was doing some further testing with the gcds-input component in an HTML form. Whenever I submit the form, the form doesn't receive the gcds-input's value. This is because the input is in the shadow DOM and the form can't see it. We will need to add a way for the component to pass that value to the form properly.

File: src/components/gcds-input/gcds-input.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+        <input
+          {...attrsInput}
+          id={inputId}
+          class={errorMessage ? 'error' : null}
+          name={inputId}
+          onBlur={this.onBlur}
+          onFocus={this.onFocus}
+          onInput={(e) => this.handleChange(e)}
+          value={inputValue ? inputValue : this.value}
+          aria-describedby={`label-for-${inputId} ${inputHint ? `hint-${inputId}` : ''} ${errorMessage ? `error-message-${inputId}` : ''}`}
```
[the remark this answers is not in the document]
Comment (melaniebmn) [reply src/components/gcds-input/gcds-input.tsx:134]: Oh weird, I've been testing with VoiceOver and it's reading out the error message.
I'll check with you on Slack!

[the remark this answers is not in the document]
Comment (melaniebmn) [reply src/components/gcds-input/gcds-input.tsx]: This should be fixed now.

File: src/components/gcds-input/gcds-input.tsx [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    };
+
+    const attrsLabel = {
+      label,
+      required,
+    }
+
+    return (
+      <Host>
+        <gcds-label
```
[the remark this answers is not in the document]
Comment (melaniebmn) [reply src/components/gcds-input/gcds-input.tsx:113]: This should be fixed now.

**Comment by melaniebmn** [comment]
> Was doing some further testing with the gcds-input component in an HTML form. Whenever I submit the form, the form doesn't receive the gcds-input's value. This is because the input is in the shadow DOM and the form can't see it. We will need to add a way for the component to pass that value to the form properly.

@ethanWallace everything should be fixed now. Could you please review it again?

**Comment by ethanWallace** [comment]
One last thing to fix up. Currently the id given to the component will be on both the `<gcds-input>` and the `<input>` tag. Will need to make sure it doesn't appear on the parent tag.

**Review by ethanWallace — APPROVED** [review]

**Pull request merged**